system/lib/wasmfs/file_table.cpp

```diff
@@ -63,20 +75,37 @@ class StderrFile : public File {
   __wasi_errno_t read(const uint8_t* buf, __wasi_size_t len) override {
     return __WASI_ERRNO_INVAL;
   };
+
+public:
+  static std::shared_ptr<StderrFile> getSharedPtr() {
+    static const std::shared_ptr<StderrFile> stderrFile = std::make_shared<StderrFile>();
+    return stderrFile;
+  }
 };
 
 FileTable::FileTable() {
-  entries.push_back(std::make_shared<OpenFileState>(0, std::make_shared<StdinFile>()));
-  entries.push_back(std::make_shared<OpenFileState>(0, std::make_shared<StdoutFile>()));
-  entries.push_back(std::make_shared<OpenFileState>(0, std::make_shared<StderrFile>()));
+  entries.push_back(std::make_shared<OpenFileState>(0, StdinFile::getSharedPtr()));
+  entries.push_back(std::make_shared<OpenFileState>(0, StdoutFile::getSharedPtr()));
+  entries.push_back(std::make_shared<OpenFileState>(0, StderrFile::getSharedPtr()));
```

Comment:
how about `getSingleton()` instead of `getSharedPtr()`? That will describe the operation at a slightly higher level.